src/common/util.h

```diff
@@ -233,3 +233,11 @@ static inline void util_setbit(uint8_t *b, uint32_t i)
 }
 #define	util_isset(a, i) isset(a, i)
 #define	util_isclr(a, i) isclr(a, i)
+
+#if defined(__GNUC__) && !defined(likely) && !defined(unlikely)
```

Comment:
What if `__GNUC__` is not defined, but `likely` and `unlikely` are... I know it's unlikely (sic!), but...
What about something like that:

```C
#if !defined(likely)
#if __GNUC__ >= 3
#define likely(x) __builtin_expect(!!(x), 1)
#else
#define likely(x) (!!(x))
#endif
#endif
```